### DISCUSSION TITLE
Type of "pd.notna" is partially unknown

### DISCUSSION THREAD

**Discussion by JanEricNitschke** [microsoft/pyright#8250] [Q&A] [answered]
I am having trouble understanding the error `Type of "notna" is partially unknown`.

The type hint is given [here](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/core/dtypes/missing.pyi#L43) and [here](https://github.com/pandas-dev/pandas-stubs/blob/main/pandas-stubs/_typing.pyi#L516).

The error occurs with any possible input, for example:

```
import pandas as pd

pd.notna(5)
```

gives the full error:

```
Type of "notna" is partially unknown
  Type of "notna" is "Overload[(obj: DataFrame) -> DataFrame, (obj: Series[Unknown]) -> Series[bool], (obj: Index[Unknown] | list[Unknown] | ExtensionArray | ndarray[Unknown, Unknown]) -> ndarray[Any, dtype[bool_]], (obj: ScalarT@notna | NaTType | NAType | None) -> TypeGuard[ScalarT@notna]]"Pylance[reportUnknownMemberType]
```

Is this an issue with the stubs or with pyright (or am i doing something wrong)?

**Comment by erictraut** [comment, marked as answer]
This is an issue with the stubs. One of the overloads for `notna` is missing a type argument. The `Series` type takes a single type argument, but it's not specified in the stub. It is therefore treated as `Unknown`.

**Reply by JanEricNitschke** [reply]
Ah, so despite that overload definitely not being applicable here it still causes that issue.

So I guess all of the type parameters would have to be specified.

For Series, but also Index, list and ndarray.

**Reply by erictraut** [reply]
Yes, the expression `pd.notna` evaluates to a type that is partially unknown. This must be evaluated prior to evaluating the call expression, which determines which overload signature is applicable. The presence of unknown type arguments can also affect overload matching.
